### ISSUE TITLE
Support JSON Pointer in @HttpExchange interface

### ISSUE THREAD

**Issue by quaff** [spring-projects/spring-framework#28576]
Sometimes we need extract data from child node instead of root document.
For example get index mappings from ElasticSearch: 
```java
@HttpExchange("http://localhost:9200")
public interface ElasticSearchIndexOperations {

	@GetExchange("/{index}/_mapping")
	@JsonPointer("/${index}/mappings")
	Mappings getMapping(@PathVariable String index);

 }
```
NOTE: placeholders in `@JsonPointer("/${index}/mappings")` need to be resolved  base on method parameters.

**Comment by rstoyanchev** [comment]
This is something we could do, but will need to isolate use of a JSON library specific APIs. 

Jackson supports [setting JsonPointer](https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/com/fasterxml/jackson/databind/ObjectReader.html) on `ObjectReader`, so we could start with that in `AbstractJackson2Decoder` if it's passed as a hint. The next question is how to pass the hint in. It can be passed to `WebClientAdapter` via `HttpRequestValues`. No good way from there to pass a hint, the closest is `ClientResponse#body(BodyExtractor)` with a decorated `BodyExtractor`, but we could provide something. One option is an overloaded method on `ClientResponse` similar to [ServerRequest](https://github.com/spring-projects/spring-framework/blob/84bc1dfa89e979232c4a36e9d75bb50b1d48453e/spring-webflux/src/main/java/org/springframework/web/reactive/function/server/ServerRequest.java#L159-L166).

**Comment by poutsma** [comment]
> * On functional API side, I think `RestClient` does not provide a way to pass hints unlike `WebClient`, could you please confirm @poutsma?

Indeed, because `RestClient` is based on the existing HttpMessageConverters, which do not support hints.

**Comment by sdeleuze** [comment]
Ah good point, we use `MappingJacksonValue` to pass JSON views, so JSON filters support could probably be added the same way.

**Comment by rstoyanchev** [comment]
> Jackson does not provide a `@JsonPointer` annotation and I am not sure Spring should introduce that as too Jackson/JSON specific (`@JsonView` is a Jackson annotation that we reuse, but I am not sure a `@JsonPointer` annotation makes sense on Jackson side).

Jackson has a `JsonPointer` type and we could take that into account if it is passed in as an argument. It should be clear it applies to the response since you shouldn't need that for the request body.

**Comment by sdeleuze** [comment]
> Jackson has a JsonPointer type and we could take that into account if it is passed in as an argument. It should be clear it applies to the response since you shouldn't need that for the request body.

I thought about that as well, but where do you take the `String` to populate it?

**Comment by sdeleuze** [comment]
Feedback after a related discussion during our team meeting:
 - For purely client side use cases, we could indeed support `JsonPointer` arguments as the related instance would be provided by the user code.
 - I remain a bit puzzled by the fact we have not identified yet a proper way to expose such feature on server-side due to the lack of `@JsonPointer` annotation, as I believe it would be equally useful for that use case, and doing that just for declarative HTTP client looks a bit niche.
